### ISSUE TITLE
DesaturatePulses.py generate exception when calling numpy min()

### ISSUE THREAD

**Issue by malfonsi** [XENON1T/pax#579]
This is detected during the processing in OSG of some of the XE1T neutron generator datasets, e.g. 9923 (zip 00047000) or 9867 (over many of the zip files).

PAX version 6.6.5 on OSG

The relevant part of the log follow:
```
Event:   5%|4         | 46/1000 [04:03<1:56:57,  7.36s/it]Welcome to cax-process
processing 170527_0757 /srv/tmp/tmp.QNgbhbvPhT/170527_0757 XENON1T
saving to output/170527_0757
Traceback (most recent call last):
  File "/cvmfs/xenon.opensciencegrid.org/releases/anaconda/2.4/envs/pax_v6.6.5_OSG/bin/cax-process", line 11, in <module>
    load_entry_point('cax==5.0.5', 'console_scripts', 'cax-process')()
  File "/cvmfs/xenon.opensciencegrid.org/releases/anaconda/2.4/envs/pax_v6.6.5_OSG/lib/python3.4/site-packages/cax-5.0.5-py3.4.egg/cax/tasks/process.py", line 458, in main
    _process(*sys.argv[1:])
  File "/cvmfs/xenon.opensciencegrid.org/releases/anaconda/2.4/envs/pax_v6.6.5_OSG/lib/python3.4/site-packages/cax-5.0.5-py3.4.egg/cax/tasks/process.py", line 159, in _process
    core.Processor(**pax_kwargs).run()
  File "/cvmfs/xenon.opensciencegrid.org/releases/anaconda/2.4/envs/pax_v6.6.5_OSG/lib/python3.4/site-packages/pax-6.6.5-py3.4.egg/pax/core.py", line 315, in run
    self.process_event(event)
  File "/cvmfs/xenon.opensciencegrid.org/releases/anaconda/2.4/envs/pax_v6.6.5_OSG/lib/python3.4/site-packages/pax-6.6.5-py3.4.egg/pax/core.py", line 276, in process_event
    event = plugin.process_event(event)
  File "/cvmfs/xenon.opensciencegrid.org/releases/anaconda/2.4/envs/pax_v6.6.5_OSG/lib/python3.4/site-packages/pax-6.6.5-py3.4.egg/pax/plugin.py", line 91, in process_event
    event = self._process_event(event)
  File "/cvmfs/xenon.opensciencegrid.org/releases/anaconda/2.4/envs/pax_v6.6.5_OSG/lib/python3.4/site-packages/pax-6.6.5-py3.4.egg/pax/plugin.py", line 108, in _process_event
    return self.transform_event(event)
  File "/cvmfs/xenon.opensciencegrid.org/releases/anaconda/2.4/envs/pax_v6.6.5_OSG/lib/python3.4/site-packages/pax-6.6.5-py3.4.egg/pax/plugins/signal_processing/DesaturatePulses.py", line 31, in transform_event
    first_saturated = _where_saturated.min()
  File "/cvmfs/xenon.opensciencegrid.org/releases/anaconda/2.4/envs/pax_v6.6.5_OSG/lib/python3.4/site-packages/numpy/core/_methods.py", line 29, in _amin
    return umr_minimum(a, axis, None, out, keepdims)
ValueError: zero-size array to reduction operation minimum which has no identity
exiting with status 255
```

**Comment by ershockley** [comment]
Here are some more runs (and corresponding zip file) that display this bug:

8515 (DM) : XENON1T-8515-000018000-000018999-000001000.zip
7805 (DM) : XENON1T-7805-000007000-000007999-000001000.zip
8462 (Cs137) : XENON1T-8462-000204000-000204999-000001000.zip
8469 (Cs137) : XENON1T-8469-000171000-000171999-000001000.zip

**Comment by feigaodm** [comment]
Should be fixed by PR(#580)

**Comment by feigaodm** [comment]
The WFs of those events which caused the bug are actually generated by the muon event which saturates lots of the channels in a very large time range. Don't know exactly why the desaturation plugin fails to correct this crazy WFs yet, but it's fixed in PR #580.  

Here are two WFs of the events, both are clearly muon events which also triggered the MV.

<img width="1260" alt="screen shot 2017-05-31 at 4 26 40 pm" src="https://cloud.githubusercontent.com/assets/18223253/26653741/24b8e9e2-4622-11e7-912c-3f52bf54c90b.png">
<img width="1264" alt="screen shot 2017-05-31 at 4 40 55 pm" src="https://cloud.githubusercontent.com/assets/18223253/26653754/2a6568c0-4622-11e7-807e-a6500178474a.png">
